lib/ansible/modules/cloud/vmware/vmware_guest_disk_facts.py

```diff
@@ -147,16 +167,64 @@ def gather_disk_facts(self, vm_obj):
                     summary=disk.deviceInfo.summary,
                     backing_filename=disk.backing.fileName,
                     backing_datastore=disk.backing.datastore.name,
-                    backing_disk_mode=disk.backing.diskMode,
-                    backing_writethrough=disk.backing.writeThrough,
-                    backing_thinprovisioned=disk.backing.thinProvisioned,
-                    backing_uuid=disk.backing.uuid,
-                    backing_eagerlyscrub=bool(disk.backing.eagerlyScrub),
                     controller_key=disk.controllerKey,
                     unit_number=disk.unitNumber,
                     capacity_in_kb=disk.capacityInKB,
                     capacity_in_bytes=disk.capacityInBytes,
                 )
+                if isinstance(disk.backing, vim.vm.device.VirtualDisk.FlatVer1BackingInfo):
```

Comment:
Just a commentary on the project's direction in general (not the PR author's) - having to transform results from the well-documented vSphere API to an arbitrary format per ansible module leads to chaos.

This style of rewriting results is unnecessary if you return the vSphere API object in json form... the vmware_guest_facts module in devel (and 2.8) can do this already, including returning complete details about the guest, including the disk backings, with all their details and without rewriting them.

Reply:
@jeking3 I totally agree with you. But before the change we didn't had a option to get facts about any VMware object from vCenter API. We can come up with strong proposal (not here of-course) and discuss in meeting so that we can arrive at roadmap kind of decision. Thanks for pointing this out.

Reply:
I'm in agreement that we should have a discussion on this. The more we keep in line with the VMware API, the easier it will be to traverse objects.